zcfan.c

```diff
@@ -130,8 +130,8 @@ static int write_fan(const char *command, const char *value) {
     return 0;
 }
 
-static void write_watchdog_timeout(const unsigned int timeout) {
-    char timeout_s[sizeof("120")]; /* max timeout value */
+static void write_watchdog_timeout(const int timeout) {
+    char timeout_s[32]; /* surely enough even for a 64-bit int in string form */
```

Comment:
120 is the cap enforced by the kernel, so it doesn't need to be this wide. Why change it?

Reply:
Changed back to sizeof("120") and added enforcement of correct values.